packages/p2p/src/stores/general-store.js

```diff
@@ -472,9 +472,19 @@ export default class GeneralStore extends BaseStore {
         );
 
         requestWS({ get_account_status: 1 }).then(({ error, get_account_status }) => {
-            const { p2p_status } = get_account_status;
+            const { authentication, p2p_poa_required, p2p_status, status } = get_account_status;
+            const { document, identity } = authentication;
             this.setIsP2PUser(p2p_status !== 'none' && p2p_status !== 'perm_ban');
 
+            if (status.includes('cashier_locked')) {
+                this.setIsBlocked(true);
+                this.hideModal();
+            } else {
+                this.setP2pPoaRequired(p2p_poa_required);
+                this.setPoaStatus(document.status);
+                this.setPoiStatus(identity.status);
+            }
+
```

Comment:
because we don't call p2p_advertiser_info if the user is not a p2p_user, better to move it in onMount, which also fixes issue where p2p_poa_required is not set